### PR TITLE
[NETBEANS-4487] Move PHP tests on Linux to GitHub Actions

### PR DESCRIPTION
**Pull request by KacerCZ**
https://issues.apache.org/jira/browse/NETBEANS-4487

Moved PHP tests on Linux from Travis to GitHub Actions.

### CODE REVIEW

**Comment by KacerCZ** [comment]
Build failed in test trying to start PHP debuging.

**Comment by junichi11** [comment]
Maybe, xdebug is needed for tests.
Need PHP? https://docs.travis-ci.com/user/languages/php/
I can be wrong.

**Comment by KacerCZ** [comment]
Unfortunately I don't have enough experience with Travis.
Travis for NetBeans is setup for Java language and this can be the reason why is not PHP/Xdebug is not available.
Moving PHP tests to GitHub Actions as discussed at https://github.com/apache/netbeans/pull/2223 can help.

@hectorespert Do you have any idea how to configure Travis for both Java and PHP languages?

**Comment by hectorespert** [comment]
> @hectorespert Do you have any idea how to configure Travis for both Java and PHP languages?

In Travis environment, probably we need to install php dependencies before run the tests.
Why you don't try to use GitHub actions for that? Seems that PHP are preconfigured.

**Comment by KacerCZ** [comment]
I created GitHub Actions job to run all PHP modules tests on Linux.
Options are copied from Travis configuration.

**Comment by KacerCZ** [comment]
OK, all tests pass with exception of PHP Debugger on Linux :(
I added xvfb wrapper to almost all steps (the exceptions being are API tests).

**Comment by hectorespert** [comment]
> OK, all tests pass with exception of PHP Debugger on Linux :(

Could you try to fix it?

**Comment by hectorespert** [comment]
It would be good to remove php tests from Travis it they run in a GitHub action.

**Comment by KacerCZ** [comment]
PHP Debugger tests are fixed - `xdebug.remote_enable=on` has to be added to php.ini.
Without it connection to Xdebug fail.

I removed PHP tests from Travis configuration.

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -46,35 +46,37 @@ jobs:
         run: ant -f platform/core.network test
 
       - name: Test PHP modules
+        env:
+          OPTS: -quiet -Dcluster.config=php -Djavac.compilerargs=-nowarn -Dbuild.compiler.deprecation=false -Dtest-unit-sys-prop.ignore.random.failures=true
```
Comment (hectorespert) [inline .github/workflows/main.yml:50]: @KacerCZ Please. Remove `-quiet -Dcluster.config=php -Djavac.compilerargs=-nowarn -Dbuild.compiler.deprecation=false` params, it isn't required.

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -98,6 +100,164 @@ jobs:
...
+          key: ${{ runner.os }}-${{ hashFiles('**/external/binaries-list') }}
+          restore-keys: ${{ runner.os }}-
+
+      - name: Setup PHP
+        uses: shivammathur/setup-php@2.3.1
+        with:
+          php-version: '7.4'
+          tools: pecl
+          extensions: xdebug
+          ini-values: xdebug.remote_enable=on
```
Comment (hectorespert) [inline .github/workflows/main.yml:126]: :+1:

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -98,6 +100,164 @@ jobs:
...
+        
+      - name: Test Hudson
+        uses: GabrielBB/xvfb-action@v1
+        with:
+          run: ant ${{ env.OPTS }} -f php/hudson.php test
+
+      - name: Test Neon
+        uses: GabrielBB/xvfb-action@v1
+        with:
+          run: ant ${{ env.OPTS }} -f php/languages.neon test
```
Comment (hectorespert) [inline .github/workflows/main.yml:142]: @KacerCZ Could you try to run Xvfb in the first steps inteads use the `GabrielBB/xvfb-action@v1` action?
For example: https://github.com/apache/netbeans/pull/2186/files
```yaml
  linux:
    name: Linux build
    runs-on: ubuntu-18.04
    env:
      DISPLAY: ":99.0"
    steps:
      - name: Setup JDK 8
        uses: actions/setup-java@v1
        with:
          java-version: '8'
          java-package: jdk+fx

      - name: Launch Xvfb
        run: Xvfb :99 -screen 0 1024x768x24 > /dev/null 2>&1 &
```

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -98,6 +100,164 @@ jobs:
 
       - name: Test Netbeans Build System
         run: ant -Dcluster.config=basic localtest
+
+  linux-php:
+    name: Linux build of PHP modules
+    runs-on: ubuntu-18.04
+    env:
+      OPTS: -quiet -Dcluster.config=php -Djavac.compilerargs=-nowarn -Dbuild.compiler.deprecation=false -Dtest-unit-sys-prop.ignore.random.failures=true
```
Comment (hectorespert) [inline .github/workflows/main.yml:108]: @KacerCZ Remove the `-quiet` and `-Djavac.compilerargs=-nowarn -Dbuild.compiler.deprecation=false` params

File: .github/workflows/main.yml (hunk 1 above)
Comment (KacerCZ) [reply .github/workflows/main.yml]: Removed.

File: .github/workflows/main.yml (hunk 4 above)
Comment (KacerCZ) [reply .github/workflows/main.yml]: Removed.

File: .github/workflows/main.yml (hunk 3 above)
Comment (KacerCZ) [reply .github/workflows/main.yml]: Changed.

**Pull request merged**